src/java/org/apache/cassandra/db/virtual/DescribeTables.java

```diff
@@ -0,0 +1,306 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.virtual;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Streams;
+
+import org.apache.cassandra.cql3.functions.Function;
+import org.apache.cassandra.cql3.functions.FunctionName;
+import org.apache.cassandra.cql3.functions.UDAggregate;
+import org.apache.cassandra.cql3.functions.UDFunction;
+import org.apache.cassandra.db.DecoratedKey;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.db.SchemaCQLHelper;
+import org.apache.cassandra.db.marshal.CompositeType;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.db.marshal.UserType;
+import org.apache.cassandra.dht.LocalPartitioner;
+import org.apache.cassandra.schema.Functions;
+import org.apache.cassandra.schema.IndexMetadata;
+import org.apache.cassandra.schema.Schema;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.schema.Types;
+import org.apache.cassandra.schema.ViewMetadata;
+import org.apache.cassandra.utils.ByteBufferUtil;
+
+public class DescribeTables
+{
+    private static final String KEYSPACE = "keyspace_name";
+    private static final String CQL = "cql";
+
+    private static final CompositeType utfComposite = CompositeType.getInstance(UTF8Type.instance, UTF8Type.instance);
+
+    public static Collection<VirtualTable> getAll(String name)
+    {
+        return ImmutableList.of(new DescribeKeyspaceTable(name),
+                                new DescribeIndexesTable(name),
+                                new DescribeTypesTable(name),
+                                new DescribeAggregatesTable(name),
+                                new DescribeFunctionsTable(name),
+                                new DescribeViewsTable(name),
+                                new DescribeTablesTable(name));
+    }
+
+    static final class DescribeKeyspaceTable extends AbstractVirtualTable
+    {
+        DescribeKeyspaceTable(String keyspace)
+        {
+            super(TableMetadata.builder(keyspace, "describe_keyspace")
+                               .comment("cql for keyspace metadata")
+                               .kind(TableMetadata.Kind.VIRTUAL)
+                               .partitioner(new LocalPartitioner(UTF8Type.instance))
+                               .addPartitionKeyColumn(KEYSPACE, UTF8Type.instance)
+                               .addRegularColumn(CQL, UTF8Type.instance)
+                               .build());
+        }
+
+        @Override
+        public DataSet data(DecoratedKey partitionKey)
+        {
+            String keyspace = UTF8Type.instance.compose(partitionKey.getKey());
```

Comment:
One thing that popped into my mind is that this requires the internal representation of the identifier instead of the cql representation.  i.e. if your keyspace name is `"MiXeDCASE"` to access this you need to query it in an unquoted manner, i.e. `select * from system_views.describe_keyspace where keyspace_name='MiXeDCASE'`.  

This is consistent with `system_schema`, but those coming from CQL, or working with quoted CQL identifiers might be surprised to see that querying with the quoted name, i.e. `'"MiXeDCASE"'` doesn't work.  That being said I think I like it the way it currently is, primarily for the consistency with `system_schema` tables.